source/LumiCalReco/src/LumiCalClusterer_getCalHits.cpp

```diff
@@ -78,7 +78,9 @@ int LumiCalClustererClass::getCalHits(	EVENT::LCEvent * evt,
 
       } else {
 	arm =(*_mydecoder)( calHitIn )["barrel"]; // from 1 and 2
-	phiCell = (*_mydecoder)( calHitIn )["phi"];
+	//FIXME: add check for phi range?
+	phiCell = (*_mydecoder)( calHitIn )["phi"]; // goes from -phiMax/2 to +phiMax/2
+	if(phiCell < 0 ) phiCell += _cellPhiMax; // need to put into positive range only
```

Comment:
What you mean "to solve this for all cases ", what are cases ? 
As I already mentioned there is unsafe processor parameter ZLayerPhiOffset, did You 
set it to zero when processing ddsim output ?
The best would be if You calculate and set this with the same method as you did
for crossingangle ( get dughters of LumiCal barrel-1  and  find the phi angle with
transformToParent method )....